File: Data Structures & Algorithms/subsets/submission-4.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []
        # subset = []
        def solve(i, subset):
            if i==len(nums):
                res.append(subset.copy())
                return

            subset.append(nums[i])
            solve(i+1, subset)
            subset.pop()
            solve(i+1, subset)

        solve(0, [])
        return res

File: Data Structures & Algorithms/find-duplicate-integer/submission-4.java
class Solution {
    public int findDuplicate(int[] nums) {
        int n = nums.length;

        for(int i=0; i<n; i++){
            nums[nums[i]%n - 1] = nums[nums[i]%n - 1] + n;
        }

        for(int i=0; i<n; i++){
            if(nums[i] / n >= 2){
                return i+1;
            }
        }

        return -1;
    }
}
